### DISCUSSION TITLE
What is "IMU Cal" and how is it used?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#250] [Questions] [answered]
The Create3 ROS2 action goal /dock includes:
```
[mobility_monitor]: Saved IMU Cal after docking with 5423.088030 seconds off the dock
```

What is "IMU Cal" and how is it used by the Create3?

Version G.3.1

**Comment by shamlian** [comment, marked as answer]
As with the previous log message, this is an internal message intended for developers. The robot performs calibration of its IMU (inertial measurement unit -- composed of the gyro and accelerometer sensors plus some software) when it is at rest. The IMU's accuracy is important for the robot's dead reckoning system.

**Reply by slowrunner** [reply]
"intended for developers" - no problem.  

Thank you for your explanation. 

I remember reading about an API for saving off an IMU calibration (in the docs for the IMU of my prior robot).  I believe it was available to use prior to and after putting the IMU into a low power "sleep" mode. 

Always looking to have a mental model of "how my robot thinks".

**Reply by shamlian** [reply]
I appreciate that -- the desire to fully understand the behaviors behind the behaviors. We tried to strike a balance between what was exposed and what was automatically taken care of, with a bias toward exposition in the logs (cryptic as it may be ;-) ), even if the user has no control of when the robot decides to do the automatic thing (for example, IMU calibration). I don't have details on, in particular, when the robot decides it's appropriate to perform this calibration, but the log will contain messages like this when it does.

**Comment by slowrunner** [comment]
closing - answered
